lib/_stream_readable.js

```diff
@@ -799,20 +800,23 @@ Readable.prototype.unpipe = function(dest) {
 // Ensure readable listeners eventually get something
 Readable.prototype.on = function(ev, fn) {
   const res = Stream.prototype.on.call(this, ev, fn);
+  const state = this._readableState;
 
   if (ev === 'data') {
+    // update readableListening
+    state.readableListening = this.listenerCount('readable') > 0;
```

Comment:
Can you change to comment above this line so that it says why we do this, rather than what we do? I assume it’s about the `this.resume()` call below?